cli/src/commands/next.rs

```diff
@@ -60,6 +62,42 @@ pub(crate) struct NextArgs {
     edit: bool,
 }
 
+pub fn choose_commit<'a>(
+    ui: &mut Ui,
+    cmd: &str,
+    commits: &'a [Commit],
+) -> Result<&'a Commit, CommandError> {
+    writeln!(ui.stdout(), "ambiguous {cmd} commit, choose one to target:")?;
+    for (i, commit) in commits.iter().enumerate() {
+        writeln!(
+            ui.stdout(),
+            "{}: [{}] {}",
+            i + 1,
+            short_commit_hash(commit.id()),
+            commit.description().lines().next().unwrap_or_default()
+        )?;
+    }
+    writeln!(ui.stdout(), "q: quit the prompt")?;
+
+    let mut choices: Vec<String> = Default::default();
+    for i in 1..(commits.len() + 1) {
+        choices.push(format!("{i}"));
+    }
+    choices.push("q".to_string());
+    let refs: Vec<&str> = choices.iter().map(|s| s.deref()).collect();
```

Comment:
nit: `s.as_ref()` or `s.as_str()` also works. Alternatively, `ui.prompt_choice()` could be generalized to accept `&[impl AsRef<str>]` or `Iterator<...>`

Reply:
I went with `&[impl AsRef<str>]`; I couldn't figure out a way to use `Iterator` that I liked.  `impl IntoIterator<Item = impl AsRef<str>>` only works well if we use it once, it doesn't work with the `loop {}`.